department-of-veterans-affairs/caseflow
# Ki/APPEALS-42563 - Package Details: Inbound Ops Team Superuser/visor: Edit Information

Resolves [APPEALS-42563 - Package Details: Inbound Ops Team Superuser/visor: Edit Information](https://jira.devops.va.gov/browse/APPEALS-42563)

# Description
This creates a new modal editCorrespondenceGeneralInformation, that allows the VADOR, Correspondence Type, and Notes to be updated. 
-Fixes some of the correspondence/correspondenceInfo issues on Correspondence Details page

## Acceptance Criteria

## Testing Plan

1. Log in as Inbound Ops Team User
2. Navigate to Correspondence Details page for Pending/Completed Correspondence
3. Click on "Package Details" tab.
4. Click on new "Edit" button.
5. Change any or all of the 3 options: VADOR, Correspondence Type, or Notes.
6. Save.
7. Verify changes on Correspondence Details Page.
### TESTING NOTES
1. The Save button should be disabled until a change is made to any of the 3 boxes.  Once a change happens, the save button will stay active, even if reverting your changes.
2. The Save action requires calling the serialized correspondences, which can take a bit.  On Save click, it should change to a loading graphic, and will close the modal automatically on completion.
3. VADOR has data validation in order to prevent future dates from being selected.

# Frontend
## User Facing Changes

### Correspondence Details with Edit Button
![Screenshot 2024-09-26 at 4 28 21 PM](https://github.com/user-attachments/assets/63b19065-b526-4a5f-8229-239ec862f080)

### Edit General Information Modal
![Screenshot 2024-09-26 at 4 28 33 PM](https://github.com/user-attachments/assets/4229de09-875e-4cfc-888d-084aaa61e934)

### Edit General Information Modal with VADOR Date Error
![Screenshot 2024-09-26 at 4 28 44 PM](https://github.com/user-attachments/assets/81761ea7-95df-42f9-a677-589690ccb8e4)

### Edit General Information Modal Saving
![Screenshot 2024-09-26 at 4 29 04 PM](https://github.com/user-attachments/assets/9d8a223c-9c9c-40ee-8db7-67f7378b78c7)

### Correspondence Details after Edit
![Screenshot 2024-09-26 at 4 29 38 PM](https://github.com/user-attachments/assets/27cf5a66-1283-463d-873c-018d7a6a10ad)


# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
